editor/assets/default_renderpipeline/builtin-dof-pass.ts

```diff
@@ -0,0 +1,323 @@
+/*
+ Copyright (c) 2021-2024 Xiamen Yaji Software Co., Ltd.
+
+ https://www.cocos.com/
+
+ Permission is hereby granted, free of charge, to any person obtaining a copy
+ of this software and associated documentation files (the "Software"), to deal
+ in the Software without restriction, including without limitation the rights to
+ use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies
+ of the Software, and to permit persons to whom the Software is furnished to do so,
+ subject to the following conditions:
+
+ The above copyright notice and this permission notice shall be included in
+ all copies or substantial portions of the Software.
+
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ THE SOFTWARE.
+*/
+
+import {
+    _decorator, assert, CCBoolean, CCFloat, CCInteger,
+    gfx, Material, renderer, rendering, Vec3, Vec4,
+} from 'cc';
+
+import { EDITOR } from 'cc/env';
+
+import {
+    BuiltinPipelineSettings,
+} from './builtin-pipeline-settings';
+
+import {
+    BuiltinPipelinePassBuilder,
+} from './builtin-pipeline-pass';
+
+import {
+    CameraConfigs,
+    getPingPongRenderTarget,
+    PipelineConfigs,
+    PipelineContext,
+} from './builtin-pipeline';
+
+const { ccclass, disallowMultiple, executeInEditMode, menu, property, requireComponent, type } = _decorator;
+
+const { Color, LoadOp, StoreOp } = gfx;
+
+export interface DofPassConfigs {
+    enableDof: boolean;
+}
+
+@ccclass('BuiltinDepthOfFieldPass')
+@menu('Rendering/BuiltinDepthOfFieldPass')
+@requireComponent(BuiltinPipelineSettings)
+@disallowMultiple
+@executeInEditMode
+export class BuiltinDepthOfFieldPass extends BuiltinPipelinePassBuilder
+    implements rendering.PipelinePassBuilder {
+    @property({
+        group: { id: 'BuiltinPass', name: 'Pass Settings', style: 'section' },
+        type: CCInteger,
+    })
+    configOrder = 0;
+    @property({
+        group: { id: 'BuiltinPass', name: 'Pass Settings', style: 'section' },
+        type: CCInteger,
+    })
+    renderOrder = 150;
+
+    @property
+    _enableDof = false;
+    @property
+    _material: Material | null = null;
+    @property
+    _minRange = 0;
+    @property
+    _maxRange = 2;
+    @property
+    _blurRadius = 1;
+    @property
+    _intensity = 1;
+    @property
+    _focusPos = new Vec3(0, 0, 0);
```

Comment:
Should these properties be public? If not, should consider mark them as `private` properties.

Reply:
Yes. You are right.